sql/core/src/main/scala/org/apache/spark/sql/execution/SparkInspectors.scala

```diff
@@ -0,0 +1,259 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution
+
+import scala.util.control.NonFatal
+
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.catalyst.{CatalystTypeConverters, InternalRow}
+import org.apache.spark.sql.catalyst.util.{ArrayData, DateTimeUtils, IntervalUtils, MapData}
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.types._
+import org.apache.spark.unsafe.types.UTF8String
+
+object SparkInspectors {
```

Comment:
`SparkInspectors` is only used for `BaseScriptTransformationExec` though, why do we need this refactoring?

Reply:
> `SparkInspectors` is only used for `BaseScriptTransformationExec` though, why do we need this refactoring?

Just like HiveInspector, handle data convert issue.  I think make it in a  separated file may be better

Reply:
Yea, `HiveInspector` is used in many places of the `hive` package, so I think it is worth a dedicated class for it. But, this class is used only for `TRANSFORM` (w/ no-serde case), so I think we don't need to make it a general class. Also, this refactoring makes the code diff hard to see and it is not related to this PR, so could you fix it in followup if necessary?

Reply:
> Yea, `HiveInspector` is used in many places of the `hive` package, so I think it is worth a dedicated class for it. But, this class is used only for `TRANSFORM` (w/ no-serde case), so I think we don't need to make it a general class. Also, this refactoring makes the code diff hard to see and it is not related to this PR, so could you fix it in followup if necessary?

Yea. I will revert this change first.

Reply:
> Yea, `HiveInspector` is used in many places of the `hive` package, so I think it is worth a dedicated class for it. But, this class is used only for `TRANSFORM` (w/ no-serde case), so I think we don't need to make it a general class. Also, this refactoring makes the code diff hard to review and it is not related to this PR, so could you fix it in followup if necessary?

I will try with json and make the logic a little simpler.